File: 4_palindromic.cpp
#include <iostream>
using namespace std;
#include <sstream>

int checkpalindrome(int number)
{

    stringstream ss;
    ss << number;
    string n;
    ss >> n;
    int length = n.length();

    for (int i = 0; i < int(length / 2); i++)
    {
        if (n[i] != n[length - i - 1])
        {
            return 0;
        }
    }

    return 1;
}

int main()
{
    int largest = 0;
    int num1, num2;
    for (int i = 999; i > 100; i--)
    {
        for (int j = 999; j > 100; j--)
        {
            // cout<<"YES";
            if (checkpalindrome(i * j))
            {

                if (i * j > largest)
                {
                    largest = i * j;
                    num1 = i;
                    num2 = j;
                }
            }
        }
    }

    cout << "Largest is:" << largest << " ";
    cout << "Product of :" << num1 << " and " << num2;
    return 0;
}

File: 4_palindromic.py
def checkpalindrome(number):
    number = str(number)

    length = len(number)
    for i in range(0, int(len(number)/2)):
        if(number[i] != number[length-i-1]):
            return False

    return True


largest = [0, 0, 0]
for i in range(999, 100, -1):
    for j in range(999, 100, -1):
        if(checkpalindrome(i*j)):
            if(i*j > largest[0]):
                largest = [i*j, i, j]

print("Largest palindromic number is {} as a product of {} and {}".format(
    largest[0], largest[1], largest[2]))
